src/deluge/model/drum/kit.cpp

```diff
@@ -994,177 +994,182 @@ void Kit::getThingWithMostReverb(Sound** soundWithMostReverb, ParamManager** par
 
-				// And if non-MPE input, just set those finger-level MPE values to 0. If an MPE instrument had been used just before, it could have left them set to something.
-				else {
-					for (int32_t i = 0; i < kNumExpressionDimensions; i++) {
-						thisDrum->lastExpressionInputsReceived[BEND_RANGE_FINGER_LEVEL][i] = 0;
-					}
-				}
+		if (!thisNoteRow || !thisNoteRow->soundingStatus) {
 
-				int16_t mpeValues[kNumExpressionDimensions];
-				thisDrum->getCombinedExpressionInputs(mpeValues);
+			if (thisNoteRow && shouldRecordNoteOn) {
 
-				// MPE stuff - if editing note, we need to take note of the initial values which might have been sent before this note-on.
-				instrumentClipView.reportMPEInitialValuesForNoteEditing(modelStackWithNoteRow, mpeValues);
+				int16_t const* mpeValuesOrNull = NULL;
 
-				if (!thisNoteRow || !thisNoteRow->soundingStatus) {
+				if (fromDevice->ports[MIDI_DIRECTION_INPUT_TO_DELUGE].isChannelPartOfAnMPEZone(channel)) {
+					mpeValuesOrNull = mpeValues;
+				}
 
-					if (thisNoteRow && shouldRecordNoteOn) {
+				instrumentClip->recordNoteOn(modelStackWithNoteRow, velocity, false, mpeValuesOrNull);
+				if (getRootUI()) {
+					getRootUI()->noteRowChanged(instrumentClip, thisNoteRow);
+				}
+			}
+			// TODO: possibly should change the MPE params' currentValue to the initial values, since that usually does get updated by the
+			// subsequent MPE that will come in. Or does that not matter?
 
-						int16_t const* mpeValuesOrNull = NULL;
+			if (thisNoteRow && thisDrum->type == DrumType::SOUND
+			    && !thisNoteRow->paramManager.containsAnyMainParamCollections()) {
+				FREEZE_WITH_ERROR("E326"); // Trying to catch an E313 that Vinz got
+			}
 
-						if (fromDevice->ports[MIDI_DIRECTION_INPUT_TO_DELUGE].isChannelPartOfAnMPEZone(channel)) {
-							mpeValuesOrNull = mpeValues;
-						}
+			beginAuditioningforDrum(modelStackWithNoteRow, thisDrum, velocity, mpeValues, channel);
+		}
+	}
 
-						instrumentClip->recordNoteOn(modelStackWithNoteRow, velocity, false, mpeValuesOrNull);
-						if (getRootUI()) {
-							getRootUI()->noteRowChanged(instrumentClip, thisNoteRow);
-						}
-					}
-					// TODO: possibly should change the MPE params' currentValue to the initial values, since that usually does get updated by the
-					// subsequent MPE that will come in. Or does that not matter?
+	// Note-off
+	else {
+		if (thisNoteRow) {
+			if (shouldRecordNotes && thisDrum->auditioned
+			    && ((playbackHandler.recording == RECORDING_ARRANGEMENT && instrumentClip->isArrangementOnlyClip())
+			        || currentSong->isClipActive(instrumentClip))) {
 
-					if (thisNoteRow && thisDrum->type == DrumType::SOUND
-					    && !thisNoteRow->paramManager.containsAnyMainParamCollections()) {
-						FREEZE_WITH_ERROR("E326"); // Trying to catch an E313 that Vinz got
+				if (playbackHandler.recording == RECORDING_ARRANGEMENT && !instrumentClip->isArrangementOnlyClip()) {}
```

Comment:
I see this is inherited but this construct is super weird.

Reply:
Yes, as far as I can tell it is a check in case of a niche situation in arranger view where the clip might not actually exist 

I couldn't think of an appropriate way to simplify it